server/server.py

```diff
@@ -99,7 +99,7 @@ def wrapper(*args, **kwargs):
             log.error('Failed to parse request data')
             ok = False
             hostname = request.remote_addr
-        if app.config['deprecated_port']:
+        if app.config.get('deprecated_port'):
```

Comment:
I think the correct fix for this is probably to remove the `if not local_only:` wrapper around this code block later in the file:

```python
    if not local_only:
        app.config['deprecated_port'] = get_port_setting(
            port, 'deprecated', False)
```